src/tmpfiles/tmpfiles.c

```diff
@@ -110,6 +110,21 @@ typedef enum ItemType {
         ADJUST_MODE = 'm', /* legacy, 'z' is identical to this */
 } ItemType;
 
+typedef enum AgeBy {
+        AGE_BY_ATIME_F = 1 << 0, /* *_F: Any file type (except directories). */
+        AGE_BY_BTIME_F = 1 << 1,
+        AGE_BY_CTIME_F = 1 << 2,
+        AGE_BY_MTIME_F = 1 << 3,
+        AGE_BY_ATIME_D = 1 << 4, /* *_D: For directories only. */
+        AGE_BY_BTIME_D = 1 << 5,
+        AGE_BY_MTIME_D = 1 << 6,
+
+        /* All file timestamp types are checked by default. */
+        AGE_BY_DEFAULT = AGE_BY_ATIME_F | AGE_BY_BTIME_F | AGE_BY_CTIME_F |
+                         AGE_BY_MTIME_F | AGE_BY_ATIME_D | AGE_BY_BTIME_D |
+                         AGE_BY_MTIME_D,
+} AgeBy;
```

Comment:
hmm, actually, thinking about this, I think it would be a lot more natural, if we'd have AgeBy with just one set of the four values, and then use it twice in the Item structure. 

i.e. drop the _F and _D stuff here, so that there are only four bits defined. Then also add:

```
AGE_BY_DEFAULT_DIRECTORY = AGE_BY_ATIME|AGE_BY_BTIME|AGE_BY_MTIME,
AGE_BY_DEFAULT_FILE = AGE_BY_ATIME|AGE_BY_BTIME|AGE_BY_CTIME|AGE_BY_MTIME,
```